sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobs.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Specifies options for listing blobs with the 
+    /// <see cref="BlobContainerClient.GetBlobsAsync"/> and
+    /// <see cref="BlobContainerClient.GetBlobsByHierarchyAsync"/>
+    /// operations.
+    /// </summary>
+    public struct GetBlobsOptions : IEquatable<GetBlobsOptions>
```

Comment:
This seems like a big struct and it's mutable. Maybe we should change it to a class.

Reply:
It's not that big once we remove `PageSizeHint`.  It gets even smaller if we change the booleans to be a `[Flags] enum`.